pkg/kapp/cmd/app/delete.go

```diff
@@ -153,12 +153,16 @@ func (o *DeleteOptions) Run() error {
 
 	err = touch.Do(func() error {
 		err := clusterChangeSet.Apply(clusterChangesGraph)
+		if err == nil && shouldFullyDeleteApp {
+			return app.Delete()
+		}
+		_, numDeleted, _ := app.GCChanges(5, nil)
```

Comment:
At this stage, `err` could be `nil` and `shouldFullyDeleteApp` could be false, and in that case, I think we should not garbage collect app-changes.
How about we just add this in the existing `err != nil` check and keep the shouldFullyDeleteApp condition?
```go
if shouldFullyDeleteApp {
    _, numDeleted, _ := app.GCChanges(5, nil)
    if numDeleted > 0 {
	o.ui.PrintLinef("Deleted %d older app changes", numDeleted)
    }
}
```

Reply:
I was thinking of the scenario when the user will partially delete the resources with the help of filters then in theory it can also lead to an enormous amount of app changes if some script is doing it (not sure whether it is much practical or not) 
Do you think should we ignore such scenarios and just garbage collect when the deletion is unsuccessful?

Reply:
After discussion with Praveen, we have decided to keep app changes the same as the deploy default which is 200 when the user partially deletes app based on some filters and max app-changes to 5 when deletion is unsuccessful due to error.